src/sidebar/services/api-routes.ts

```diff
@@ -0,0 +1,69 @@
+import type {
+  IndexResponse,
+  LinksResponse,
+  RouteMap,
+  RouteMetadata,
+} from '../../types/api';
+import type { SidebarSettings } from '../../types/config';
+import { fetchJSON } from '../util/fetch';
+import { retryPromiseOperation } from '../util/retry';
+
+/**
+ * Fetch an API metadata file, retrying the operation if it fails.
+ */
+function getJSON<T>(url: string): Promise<T> {
+  return retryPromiseOperation(
+    () =>
+      // nb. The `/api/` and `/api/links` routes are fetched without specifying
+      // any additional headers/config so that we can use `<link rel="preload">` in
+      // the `/app.html` response to fetch them early, while the client JS app
+      // is loading.
+      fetchJSON(url) as Promise<T>
+  );
+}
+
+/**
+ * A service which fetches and caches API route metadata.
+ */
+// @inject
+export class APIRoutesService {
+  private _apiURL: SidebarSettings['apiUrl'];
```

Comment:
I think I would prefer just to type this as `string` and avoid the indirection for the reader of needing to go and look up what this field is.

Reply:
I used this approach to keep types "in sync", but considering it's just `string`, I agree with your point.

For a more complex type, where you have to go to some definition anyway, might make more sense to do this.